resources/nats-streaming/templates/pre-install-hook.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-ns-check
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Release.Name }}-ns-check
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ .Release.Name }}-ns-check
+  labels:
+    app: {{ .Release.Name }}-ns-check
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+rules:
+  - apiGroups: [""]
+    resources: ["namespaces"]
+    verbs: ["create","get","list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Release.Name }}-ns-check
+  labels:
+    app: {{ .Release.Name }}-ns-check
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ .Release.Name }}-ns-check
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Release.Name }}-ns-check
+    namespace: {{ .Release.Namespace }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: pre-install-job
+  annotations:
+    helm.sh/hook: "pre-install"
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    sidecar.istio.io/inject: "false"
+spec:
+  backoffLimit: 1
+  template:
+    metadata:
```

Comment:
(nit) can we have a label for the container executing the job? For uninstallation we have [this one](https://github.com/kyma-project/kyma/blob/master/resources/knative-eventing/charts/knative-eventing/templates/uninstall-eventing.yaml#L829=

Reply:
Done